api/src/test/java/org/openmrs/api/impl/ConditionServiceImplTest.java

```diff
@@ -0,0 +1,190 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.impl;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.openmrs.*;
+import org.openmrs.api.ConditionService;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.context.Context;
+import org.openmrs.test.BaseContextSensitiveTest;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Unit tests for methods that are specific to the {@link ConditionServiceImpl}. General tests that
+ * would span implementations should go on the {@link ConditionService}.
+ */
+public class ConditionServiceImplTest extends BaseContextSensitiveTest {
+	
+	private final String UUID = "08002000-4469-12q3-551f-0339000c9a76";
+	protected static final String CONDITION_XML = "org/openmrs/api/include/ConditionServiceImplTest-SetupCondition.xml";
+	
+	private ConditionService conditionService;
+	
+	private PatientService patientService;
+	
+	@Rule
+	public ExpectedException expectedException = ExpectedException.none();
+	
+	@Before
+	public void setup (){
+		if(conditionService == null){
+			conditionService = Context.getConditionService();
+		}
+		if(patientService == null){
+			patientService = Context.getPatientService();
+		}
+		executeDataSet(CONDITION_XML);
+	}
+
+	/**
+	 * @see ConditionService#saveCondition(Condition) 
+	 */
+	@Test
+	public void saveCondition_shouldSaveNewCondition(){
+		int patientId = 2;
+		Date date = new Date();
+		Condition condition = new Condition();
+		condition.setCondition(new CodedOrFreeText());
+		condition.setClinicalStatus(ConditionClinicalStatus.ACTIVE);
+		condition.setAdditionalDetail("saving a condition");
+		condition.setVerificationStatus(ConditionVerificationStatus.CONFIRMED);
+		condition.setOnsetDate(date);
+		condition.setUuid(UUID);
+		condition.setPatient(patientService.getPatient(patientId));
+		expectedException = ExpectedException.none();
+		conditionService.saveCondition(condition);
+		Condition savedCondition = conditionService.getConditionByUuid(UUID);
+		Assert.assertEquals(ConditionVerificationStatus.CONFIRMED, savedCondition.getVerificationStatus());
+		Assert.assertEquals(ConditionClinicalStatus.ACTIVE, savedCondition.getClinicalStatus());
+		Assert.assertEquals(UUID, savedCondition.getUuid());
+	}
+
+	/**
+	 * @see ConditionService#getConditionByUuid(String)
+	 */
+	@Test
+	public void getConditionByUuid_shouldFindConditionGivenValidUuid() {
+		executeDataSet(CONDITION_XML);
```

Comment:
As for executeDataSet() and the methods below, don't you already do this in setup()?